### PR TITLE
Add snapshot testing for `NetworkConfig`, `GenesisConfig` and `Genesis`

### PR DESCRIPTION
**Pull request by mwtian**
Add snapshot testing for a few "root" config structures: `NetworkConfig`, `GenesisConfig` and `Genesis`.

I'm not 100% sure about the utility of snapshotting serialized empty `GenesisConfig`.

Also, it seems better to support creating configs with specific port numbers, instead of clearing them for the snapshot tests. I imagine in containers, having stable ports would improve usability too. I can do the refactoring in another PR.

### CODE REVIEW

**Comment by punwai** [comment]
Please forcepush to this PR :(

File: crates/sui-config/tests/snapshot_tests.rs
Comment (huitseeker) [inline crates/sui-config/tests/snapshot_tests.rs:7]: I would expand this comment to give a bit more context about this, and list:
- that if you broke one of those tests, you probably have a real impact on a production configuration file, you should call this out, and mark your PR as a breaking change.
- if you are managing a production configuration file, you should add yourself to the CODEOWNERS of the corresponding snapshot tests.
- if you are changing one of those snapshots, you should wait for a review from the CODEOWNER (if any) before merging.

File: crates/sui-config/tests/snapshot_tests.rs
Comment (huitseeker) [inline crates/sui-config/tests/snapshot_tests.rs:30]: You can actually get an `ObjectID` from an address, and in fact I think the best way to get it is to use `ObjectID::random()`.

**Review by huitseeker — COMMENTED** [review]
This looks pretty darn good to me, thanks for the fast turnaround.
I think the main aspects are:
- pedagogy on what to do if tests are broken,
- slightly better `ObjectID` derivation.

File: crates/sui-config/tests/snapshot_tests.rs
Comment (mwtian) [reply crates/sui-config/tests/snapshot_tests.rs:17]: Reworded here.

File: crates/sui-config/tests/snapshot_tests.rs
Comment (mwtian) [reply crates/sui-config/tests/snapshot_tests.rs]: Changed to getting `ObjectID` from the fake address. The object ID needs to be static because it is part of the snapshot.

Comment (huitseeker) [reply crates/sui-config/tests/snapshot_tests.rs]: Of course! makes sense.

**Review by huitseeker — APPROVED** [review]
This is fantastic! Thank you so much for this contribution @mwtian !!!

**Comment by mwtian** [comment]
Thanks @huitseeker!

**Pull request merged**
